engine/gameobject/src/gameobject/gameobject.cpp

```diff
@@ -1033,9 +1035,15 @@ namespace dmGameObject
         return RESULT_OK;
     }
 
-    Result SetIdentifier(HCollection hcollection, HInstance instance, dmhash_t id)
+    Result SetIdentifier(HCollection collection, HInstance instance, dmhash_t id)
     {
-        return SetIdentifier(hcollection->m_Collection, instance, id);
+        return SetIdentifier(collection->m_Collection, instance, id);
+    }
+
+    Result SetConstructedIdentifier(HCollection collection, HInstance instance)
+    {
+        instance->m_Generated = true;
+        return SetIdentifier(collection->m_Collection, instance, ConstructInstanceId());
```

Comment:
If we generate ID, we want to mark the instance as `m_Generated`, so it will release hash when object will be removed (we already have this functionality which used only for collection factories)